.github/workflows/main.yml

```diff
@@ -26,11 +26,23 @@ jobs:
 
       - name: Use OCaml ${{ matrix.ocaml-compiler }}
         uses: ocaml/setup-ocaml@v2
+        if: runner.os != 'Windows'
         with:
           ocaml-compiler: ${{ matrix.ocaml-compiler }}
           dune-cache: ${{ matrix.os == 'ubuntu-latest' }}
           opam-depext-flags: --with-test
 
+      - name: Use OCaml ${{ matrix.ocaml-compiler }}
+        uses: ocaml/setup-ocaml@v2
+        if: runner.os == 'Windows'
+        with:
+          ocaml-compiler: ${{ matrix.ocaml-compiler }}
+          dune-cache: false
+          opam-depext-flags: --with-test
+          opam-repositories: |
+            windows: https://github.com/fdopen/opam-repository-mingw.git
+            default: https://github.com/ocaml/opam-repository.git
```

Comment:
`js_of_ocaml` 4.1.0 is not yet available on the opam windows fork, so we fallback to the official opam repo